dask_planner/src/sql/optimizer.rs

```diff
@@ -34,13 +34,15 @@ use filter_columns_post_join::FilterColumnsPostJoin;
 /// and their ordering in regards to their impact on the underlying `LogicalPlan` instance
 pub struct DaskSqlOptimizer {
     skip_failing_rules: bool,
+    #[allow(dead_code)]
+    statistics: HashMap<String, f64>,
     optimizer: Optimizer,
 }
 
 impl DaskSqlOptimizer {
     /// Creates a new instance of the DaskSqlOptimizer with all the DataFusion desired
     /// optimizers as well as any custom `OptimizerRule` trait impls that might be desired.
-    pub fn new(skip_failing_rules: bool) -> Self {
+    pub fn new(skip_failing_rules: bool, statistics: HashMap<String, f64>) -> Self {
         let rules: Vec<Arc<dyn OptimizerRule + Sync + Send>> = vec![
             Arc::new(InlineTableScan::new()),
```

Comment:
With these changes, we can pass table statistics into optimizer rules. For example, with https://github.com/dask-contrib/dask-sql/pull/1027 we could do something like `Arc::new(JoinReorder::new(2, 0.3, true, 1.0, statistics.clone()))` where `statistics` is just another one of the fields that we're passing into a new instance of the `JoinReorder` struct.